### ISSUE TITLE
LFS v2 cannot fetch files from the server.

### ISSUE THREAD

**Issue by midworld** [git-lfs/git-lfs#2029]
Hi,
I upgraded git-lfs v1.5.6 to v2.0.0/v2.0.1 but LFS v2 cannot fetch objects from the server.

Here is my config file:
```
$ cat .lfsconfig 
[lfs]
	url = http://git.mydomain.net:9997/dev/bh-cli
```

and result of git env:
```
$ git lfs env
git-lfs/2.0.1 (GitHub; darwin amd64; go 1.8)
git version 2.12.0

Endpoint=http://git.mydomain.net:9997/dev/bh-cli (auth=basic)
LocalWorkingDir=/Users/midworld/Documents/works/test/bh-cli
LocalGitDir=/Users/midworld/Documents/works/test/bh-cli/.git
LocalGitStorageDir=/Users/midworld/Documents/works/test/bh-cli/.git
LocalMediaDir=/Users/midworld/Documents/works/test/bh-cli/.git/lfs/objects
LocalReferenceDir=
TempDir=/Users/midworld/Documents/works/test/bh-cli/.git/lfs/tmp
ConcurrentTransfers=3
TusTransfers=false
BasicTransfersOnly=false
SkipDownloadErrors=false
FetchRecentAlways=false
FetchRecentRefsDays=7
FetchRecentCommitsDays=0
FetchRecentRefsIncludeRemotes=true
PruneOffsetDays=3
PruneVerifyRemoteAlways=false
PruneRemoteName=origin
AccessDownload=basic
AccessUpload=basic
DownloadTransfers=basic
UploadTransfers=basic
git config filter.lfs.process = "git-lfs filter-process"
git config filter.lfs.smudge = "git-lfs smudge -- %f"
git config filter.lfs.clean = "git-lfs clean -- %f"
```

But LFS v2 fail fetching files:
```
$ git lfs fetch
Fetching develop
Git LFS: (0 of 373 files) 0 B / 190.13 MB
batch response: creds: parse git@git.mydomain.net:bh/bh-cli.git: first path segment in URL cannot contain colon
batch response: creds: parse git@git.mydomain.net:bh/bh-cli.git: first path segment in URL cannot contain colon
batch response: creds: parse git@git.mydomain.net:bh/bh-cli.git: first path segment in URL cannot contain colon
batch response: creds: parse git@git.mydomain.net:bh/bh-cli.git: first path segment in URL cannot contain colon
Warning: errors occurred
```

This git lfs server is git-lfs-test-server and it works perfectly with git-lfs v1.5.6. I think it is a bug on parsing git lfs server URL.

Any idea for this issue?

**Comment by ttaylorr** [comment]
Hi @midworld thanks for opening, and sorry that you're having trouble. The error that you're seeing "first path segment ..." comes from the `net/url.Parse` function in the Go standard library. Particularly, it was introduced in https://github.com/golang/go/commit/c5ccbdd22bdbdc43d541b7e7d4ed66ceb559030e, which was tagged for the first time in `go1.8`.

Since the 1.5.x series (and below) of LFS was compiled against various versions of Go 1.7.x, this error is new to LFS.

I think to fix this, [this][1] function would have to be updated to strip the colon from the URL.

[1]: https://github.com/git-lfs/git-lfs/blob/v2.0.1/lfsapi/endpoint_finder.go#L144-L162

**Comment by ttaylorr** [comment]
What I'm more curious in is that you have a `.lfsconfig` in the root of your repository (with a valid `lfs.url`) but LFS seems not to be using it.

**Comment by midworld** [comment]
I think LFS is using it. When I tried with the following config file:
```
$ cat .lfsconfig 
[lfs]
	url = http://git.mydomain.net/dev/bh-cli
```
It failed with correct errors.
```
$ git lfs fetch
Fetching develop
Git LFS: (0 of 373 files) 0 B / 190.13 MB
                                                           
batch response: Repository or object not found: http://git.mydomain.net/dev/bh-cli/objects/batch
Check that it exists and that you have proper access to it
batch response: Repository or object not found: http://git.mydomain.net/dev/bh-cli/objects/batch
Check that it exists and that you have proper access to it
batch response: Repository or object not found: http://git.mydomain.net/dev/bh-cli/objects/batch
Check that it exists and that you have proper access to it
batch response: Repository or object not found: http://git.mydomain.net/dev/bh-cli/objects/batch
Check that it exists and that you have proper access to it
Warning: errors occurred
```

**Comment by FeodorFitsner** [comment]
Hi guys,

When do you plan to release 2.0.2? Or is it safer to stick to 1.5.6 for now?

**Comment by dheater** [comment]
I had the same problem with 2.0.1. Dropped back to version 1.5.6 and it's working.

**Comment by technoweenie** [comment]
I'd like to get v2.0.2 out next week if possible. Here's the milestone: https://github.com/git-lfs/git-lfs/issues?q=is%3Aopen+is%3Aissue+milestone%3Av2.0.2

**Comment by FeodorFitsner** [comment]
Sounds great, thank you!

**Comment by rjbell4** [comment]
FWIW, as a working, a ssh:// URL seems to work.

**Comment by ttaylorr** [comment]
👋  all, thanks for opening this issue again. I just posted #2090, which I'm hoping to backport into the `release-2.0` tree to be included in `2.0.2` sometime within the next few days. I've attached a build that has this patch applied to it.

If anyone wouldn't mind trying it out to see if it fixes this issue, I'd be greatly appreciative 🙇.

[git-lfs-darwin-386-2.1.0-pre.tar.gz](https://github.com/git-lfs/git-lfs/files/874088/git-lfs-darwin-386-2.1.0-pre.tar.gz)
[git-lfs-darwin-amd64-2.1.0-pre.tar.gz](https://github.com/git-lfs/git-lfs/files/874089/git-lfs-darwin-amd64-2.1.0-pre.tar.gz)
[git-lfs-freebsd-386-2.1.0-pre.tar.gz](https://github.com/git-lfs/git-lfs/files/874090/git-lfs-freebsd-386-2.1.0-pre.tar.gz)
[git-lfs-freebsd-amd64-2.1.0-pre.tar.gz](https://github.com/git-lfs/git-lfs/files/874086/git-lfs-freebsd-amd64-2.1.0-pre.tar.gz)
[git-lfs-linux-386-2.1.0-pre.tar.gz](https://github.com/git-lfs/git-lfs/files/874087/git-lfs-linux-386-2.1.0-pre.tar.gz)
[git-lfs-linux-amd64-2.1.0-pre.tar.gz](https://github.com/git-lfs/git-lfs/files/874091/git-lfs-linux-amd64-2.1.0-pre.tar.gz)
[git-lfs-windows-386-2.1.0-pre.zip](https://github.com/git-lfs/git-lfs/files/874092/git-lfs-windows-386-2.1.0-pre.zip)
[git-lfs-windows-amd64-2.1.0-pre.zip](https://github.com/git-lfs/git-lfs/files/874093/git-lfs-windows-amd64-2.1.0-pre.zip)

**Comment by jasonwhite** [comment]
@ttaylorr Thanks. I just tested it on a large repository and it fixes the error for me!

```
$ git lfs env
git-lfs/2.1.0-pre (GitHub; windows amd64; go 1.8; git 0ee430bb)
git version 2.12.2.windows.1
```
